### PR TITLE
Working toward object_reform

### PR DESCRIPTION
**Pull request by gkoz**
* Move the generated files files to `mod auto` and generate `auto/mod.rs`
* Mangle module, function, parameter names if they conflict with Rust keywords
* Generate non-widget objects
* Add `Bitfield` and `Filename` support
* Minor internal and non-functional output tweaks

### CODE REVIEW

**Comment by EPashkin** [comment]
Maybe add Popover (or other) as example of widget version?
```
[[object]]
name = "Gtk.Popover"
status = "generate"
```

**Comment by gkoz** [comment]
I've add a lot of widgets to my local copy in a branch of the `gtk` repo, just not sure what should or shouldn't go into the config here...

**Comment by gkoz** [comment]
I could push one that build cleanly on the object_reform branch, or one that exposes some generator bugs.

**Comment by EPashkin** [comment]
Understand

File: src/analysis/functions.rs [hunk 1]
```diff
@@ -60,6 +63,12 @@ fn analyze_function(env: &Env, func: &library::Function, class_tid: library::Typ
         }
     }
 
+    if !commented {
+        for s in used_types {
+            all_used_types.insert(s);
+        }
+    }
```
Comment (gkoz) [inline src/analysis/functions.rs]: This  might need to be revisited when we encounter unused import warnings depending on the set cfg versions.

An alternative is putting something like this in modules that have version cfgs:
```
#![cfg_attr(not(gtk_3_10), allow(unused_imports))]
```

**Comment by gkoz** [comment]
The "Analyzing... Generating..." output to stdout is getting out of hand, I'd prefer to switch to logging macros, so e.g. ignoring a class would have higher severity than just generating one and easier to notice.

**Comment by EPashkin** [comment]
Agree

**Comment by gkoz** [comment]
I've addressed the "generate mod.rs in separate function" comment.

Going to merge this.

**Comment by EPashkin** [comment]
Thanks, i rebase my PR after merging (it will have conflict)

**Comment by gkoz** [comment]
@homu r+

**Pull request merged**